### PR TITLE
SK-865 development delete by skyflow id in python sdk

### PR DESCRIPTION
**Pull request by skyflow-lipsa**

### CODE REVIEW

**Review by skyflow-vivek — APPROVED** [review]

File: skyflow/errors/_skyflow_errors.py [hunk 1]
```diff
@@ -43,11 +44,18 @@ class SkyflowErrorMessages(Enum):
     INVALID_RECORDS_TYPE = "Records key has value of type %s, expected list"
     INVALID_FIELDS_TYPE = "Fields key has value of type %s, expected dict"
     INVALID_TABLE_TYPE = "Table key has value of type %s, expected string"
+    INVALID_TABLE_TYPE_DELETE = "Table of type string is required in records array"
     INVALID_IDS_TYPE = "Ids key has value of type %s, expected list"
     INVALID_ID_TYPE = "Id key has value of type %s, expected string"
+    INVALID_ID_TYPE_DELETE = "Id of type string is required at index 0 in records array"
```
Comment (skyflow-vivek) [inline skyflow/errors/_skyflow_errors.py:50]: It shouldn't be hard coded as `index 0`. For scenarios where `id` is present 0th record, but not in nth record, Error message will be inconsistent  It should be `index %s` instead where index is populated dynamically.

File: skyflow/errors/_skyflow_errors.py [hunk 2]
```diff
@@ -43,11 +44,18 @@ class SkyflowErrorMessages(Enum):
...
     INVALID_REDACTION_TYPE = "Redaction key has value of type %s, expected Skyflow.Redaction"
     INVALID_COLUMN_NAME = "Column name has value of type %s, expected string"
     INVALID_COLUMN_VALUE = "Column values has value of type %s, expected list"
+    INVALID_RECORDS_IN_DELETE = "Invalid records. records object should be an array"
+    EMPTY_RECORDS_IN_DELETE = "records array cannot be empty"
+    EMPTY_ID_IN_DELETE = "Id cannot be empty in records array"
```
Comment (skyflow-vivek) [inline skyflow/errors/_skyflow_errors.py:56]: For consistency across SDKs, you should add `index` here to indicate in which record `id` key is missing.

File: skyflow/errors/_skyflow_errors.py [hunk 3]
```diff
@@ -43,11 +44,18 @@ class SkyflowErrorMessages(Enum):
...
+    EMPTY_TABLE_IN_DELETE = "Table cannot be empty in records array"
```
Comment (skyflow-vivek) [inline skyflow/errors/_skyflow_errors.py:57]: For consistency across SDKs, you should add `index` here to indicate in which record `table` key is missing.

File: skyflow/vault/_client.py [hunk 4]
```diff
@@ -172,4 +174,83 @@ def update(self, updateInput, options: UpdateOptions = UpdateOptions()):
...
+            elif record_list == "":
+                error.update({"error": {"code": SkyflowErrorCodes.INVALID_INPUT.value,
+                                        "description": SkyflowErrorMessages.EMPTY_ID_IN_DELETE.value}})
+                return error
+        except KeyError:
+            error.update({"error": {"code": SkyflowErrorCodes.INVALID_INDEX.value,
+                                    "description": SkyflowErrorMessages.IDS_KEY_ERROR.value}})
+            return error
+        try:
+            record_table = records["records"][0]['table']
```
Comment (skyflow-vivek) [inline skyflow/vault/_client.py:228]: You shouldn't hard code the `index` of records here. It should be a loop over the records array. We need to validate every record in records not just the record at 0th index!

File: skyflow/errors/_skyflow_errors.py (hunk 1 above)
Comment (skyflow-lipsa) [reply skyflow/errors/_skyflow_errors.py:50]: okay

File: tests/vault/test_delete_by_id.py [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+        invalidData = {"records": "invalid_data"}
+        result = self.client.delete_by_id(invalidData)
+        self.assertEqual(result, {
+            "error": {
+                "code": SkyflowErrorCodes.INVALID_INPUT.value,
+                "description": SkyflowErrorMessages.INVALID_RECORDS_IN_DELETE.value
+            }
+        })
+
+    def testDeleteByIdEmptyRecordsListType(self):
```
[the remark this answers is not in the document]
Comment (skyflow-lipsa) [reply tests/vault/test_delete_by_id.py:103]: okay

**Review by skyflow-vivek — APPROVED** [review]

**Review by skyflow-bharti — APPROVED** [review]

**Pull request merged**